fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexExpressions.java

```diff
@@ -215,6 +216,10 @@ public static <T extends RecordSource<T>> void getFieldsRecursively(@Nonnull Key
                                                                         @Nonnull T source, @Nonnull DocumentDestination<T> destination,
                                                                         @Nullable String fieldNamePrefix, int keyIndex, int groupingCount,
                                                                         @Nonnull List<Integer> overriddenKeyRanges) {
+        if (expression instanceof RecordTypeKeyExpression) {
+            return;
```

Comment:
Why does this just return immediately on a `RecordTypeKeyExpression`? I would have though that this would need to return the associated record type's value of the record, but maybe that's also not quite right. It might actually be more of a suggestion that the logic here for extracting the copier for primary key expressions isn't quite right (e.g., it should share more code with the normal tuple-to-copier code for VALUE indexes instead of the Lucene index code)

Reply:
Added comment